### PR TITLE
issue 5851 support jib client

### PR DESCRIPTION
**Pull request by mmpei**
Signed-off-by: mmpei <peimingming1986@126.com>

add for supporting jib client.
test success.
step1.compile harbor and bring up
2.config new harbor IP and account password in pom.xml in jib block.
3.compile using jib plugin and jib will auto pushing image to harbor.
4.check the image in project in harbor. success.

### CODE REVIEW

**Comment by steven-zou** [comment]
@mmpei 

Please see my comments!

**Comment by reasonerjt** [comment]
Additionally it'd better to add a `checkEvent` or `checkRequst` func to check if the http request of given method or user-agent should be handled, and write a UT to it.

**Comment by steven-zou** [comment]
@mmpei 

Any updates of this PR?

**Comment by mmpei** [comment]
i will fix this gofmt issue.

File: src/ui/service/notifications/registry/handler.go [hunk 1]
```diff
@@ -159,7 +159,7 @@ func filterEvents(notification *models.Notification) ([]*models.Event, error) {
 		}
 
 		// pull and push manifest by docker-client or vic
-		if (strings.HasPrefix(event.Request.UserAgent, "docker") || strings.HasPrefix(event.Request.UserAgent, vicPrefix)) &&
+		if (strings.HasPrefix(event.Request.UserAgent, "docker") || strings.HasPrefix(event.Request.UserAgent, vicPrefix) || strings.HasPrefix(event.Request.UserAgent, "jib")) &&
 			(event.Action == "pull" || event.Action == "push") {
 			events = append(events, &event)
 			log.Debugf("add event to collect: %s", event.ID)
```
Comment (steven-zou) [inline src/ui/service/notifications/registry/handler.go]: "add event to collect: %s" should be "add event to collection: %s"?

File: src/ui/service/notifications/registry/handler.go [hunk 2]
```diff
@@ -159,7 +159,7 @@ func filterEvents(notification *models.Notification) ([]*models.Event, error) {
 		}
 
 		// pull and push manifest by docker-client or vic
-		if (strings.HasPrefix(event.Request.UserAgent, "docker") || strings.HasPrefix(event.Request.UserAgent, vicPrefix)) &&
+		if (strings.HasPrefix(event.Request.UserAgent, "docker") || strings.HasPrefix(event.Request.UserAgent, vicPrefix) || strings.HasPrefix(event.Request.UserAgent, "jib")) &&
```
Comment (steven-zou) [inline src/ui/service/notifications/registry/handler.go]: Prefix checking and action checking can be abstracted to a separate handler/function?

e.g:

  `EventHandler.AcceptAgents(...string).AcceptActions(...string).Handle(event)?`

This is just a suggestion, you can try your own one.

**Comment by steven-zou** [comment]
@mmpei 

Additionally, conflicts happened, please try to rebase the code after refactoring. Thanks.

**Comment by steven-zou** [comment]
@mmpei 

Any updates?

**Comment by mmpei** [comment]
@steven-zou   update, please check again.

**Comment by mmpei** [comment]
see https://github.com/goharbor/harbor/issues/6086
i also fixed this issue. please check.

**Comment by steven-zou** [comment]
@mmpei 

your commits should be squashed. Two many commits in the PR.

**Comment by YangGuang001** [comment]
@mmpei 
this is ui.log
```
Nov 27 13:59:39 172.24.0.1 ui[1565]: 2018/11/27 13:59:39 #033[1;44m[D] [server.go:2619] |     172.24.0.6|#033[42m 200 #033[0m|      234.6µs|   match|#033[46m POST    #033[0m /service/notifications   r:/service/notifications#033[0m
Nov 27 13:59:43 172.24.0.1 ui[1565]: 2018/11/27 13:59:43 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|        240µs|   match|#033[44m GET     #033[0m /api/ping   r:/api/ping#033[0m
Nov 27 13:59:46 172.24.0.1 ui[1565]: 2018/11/27 13:59:46 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|    13.8419ms|   match|#033[44m GET     #033[0m /     r:/#033[0m
Nov 27 13:59:58 172.24.0.1 ui[1565]: 2018/11/27 13:59:58 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 202 #033[0m|  55.0750339s|   match|#033[42m PATCH   #033[0m /v2/ysf/foundation/blobs/uploads/49ee6750-793d-4caa-a0b9-0f47b3c13973   r:/v2/*#033[0m
Nov 27 13:59:58 172.24.0.1 ui[1565]: 2018/11/27 13:59:58 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 201 #033[0m|      78.55ms|   match|#033[43m PUT     #033[0m /v2/ysf/foundation/blobs/uploads/49ee6750-793d-4caa-a0b9-0f47b3c13973   r:/v2/*#033[0m
Nov 27 13:59:58 172.24.0.1 ui[1565]: 2018/11/27 13:59:58 #033[1;44m[D] [server.go:2619] |     172.24.0.6|#033[42m 200 #033[0m|        337µs|   match|#033[46m POST    #033[0m /service/notifications   r:/service/notifications#033[0m
Nov 27 14:00:13 172.24.0.1 ui[1565]: 2018/11/27 14:00:13 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|    12.0136ms|   match|#033[44m GET     #033[0m /api/ping   r:/api/ping#033[0m
Nov 27 14:00:16 172.24.0.1 ui[1565]: 2018/11/27 14:00:16 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|      465.1µs|   match|#033[44m GET     #033[0m /     r:/#033[0m
Nov 27 14:00:43 172.24.0.1 ui[1565]: 2018/11/27 14:00:43 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|    14.2556ms|   match|#033[44m GET     #033[0m /api/ping   r:/api/ping#033[0m
Nov 27 14:00:46 172.24.0.1 ui[1565]: 2018/11/27 14:00:46 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|     2.4908ms|   match|#033[44m GET     #033[0m /     r:/#033[0m
Nov 27 14:00:47 172.24.0.1 ui[1565]: 2018/11/27 14:00:47 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 202 #033[0m| 1m45.911453s|   match|#033[42m PATCH   #033[0m /v2/ysf/foundation/blobs/uploads/b53a874f-e7ab-4cee-8ec8-1d0013c00e3e   r:/v2/*#033[0m
Nov 27 14:00:47 172.24.0.1 ui[1565]: 2018/11/27 14:00:47 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 201 #033[0m|   116.9164ms|   match|#033[43m PUT     #033[0m /v2/ysf/foundation/blobs/uploads/b53a874f-e7ab-4cee-8ec8-1d0013c00e3e   r:/v2/*#033[0m
Nov 27 14:00:47 172.24.0.1 ui[1565]: 2018/11/27 14:00:47 #033[1;44m[D] [server.go:2619] |     172.24.0.6|#033[42m 200 #033[0m|      255.5µs|   match|#033[46m POST    #033[0m /service/notifications   r:/service/notifications#033[0m
Nov 27 14:01:13 172.24.0.1 ui[1565]: 2018/11/27 14:01:13 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|     8.5536ms|   match|#033[44m GET     #033[0m /api/ping   r:/api/ping#033[0m
Nov 27 14:01:17 172.24.0.1 ui[1565]: 2018/11/27 14:01:17 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|      366.3µs|   match|#033[44m GET     #033[0m /     r:/#033[0m
Nov 27 14:01:44 172.24.0.1 ui[1565]: 2018/11/27 14:01:44 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|    10.1392ms|   match|#033[44m GET     #033[0m /api/ping   r:/api/ping#033[0m
Nov 27 14:01:47 172.24.0.1 ui[1565]: 2018/11/27 14:01:47 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|      474.4µs|   match|#033[44m GET     #033[0m /     r:/#033[0m
Nov 27 14:02:01 172.24.0.1 ui[1565]: 2018/11/27 14:02:01 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 202 #033[0m| 3m0.1222305s|   match|#033[42m PATCH   #033[0m /v2/ysf/foundation/blobs/uploads/53901421-5b08-4d67-b79b-d6fb1abfd91f   r:/v2/*#033[0m
Nov 27 14:02:01 172.24.0.1 ui[1565]: 2018/11/27 14:02:01 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 201 #033[0m|   129.8625ms|   match|#033[43m PUT     #033[0m /v2/ysf/foundation/blobs/uploads/53901421-5b08-4d67-b79b-d6fb1abfd91f   r:/v2/*#033[0m
Nov 27 14:02:01 172.24.0.1 ui[1565]: 2018/11/27 14:02:01 #033[1;44m[D] [server.go:2619] |     172.24.0.6|#033[42m 200 #033[0m|        305µs|   match|#033[46m POST    #033[0m /service/notifications   r:/service/notifications#033[0m
Nov 27 14:02:01 172.24.0.1 ui[1565]: 2018/11/27 14:02:01 #033[1;44m[D] [server.go:2619] |     172.24.0.6|#033[42m 200 #033[0m|     1.1805ms|   match|#033[46m POST    #033[0m /service/notifications   r:/service/notifications#033[0m
Nov 27 14:02:01 172.24.0.1 ui[1565]: 2018/11/27 14:02:01 #033[1;44m[D] [server.go:2619] |     172.24.0.6|#033[42m 200 #033[0m|     1.2502ms|   match|#033[46m POST    #033[0m /service/notifications   r:/service/notifications#033[0m
Nov 27 14:02:01 172.24.0.1 ui[1565]: 2018/11/27 14:02:01 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 201 #033[0m|   257.3954ms|   match|#033[43m PUT     #033[0m /v2/ysf/foundation/manifests/latest   r:/v2/*#033[0m
Nov 27 14:02:01 172.24.0.1 ui[1565]: 2018/11/27 14:02:01 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 201 #033[0m|   274.3744ms|   match|#033[43m PUT     #033[0m /v2/ysf/foundation/manifests/4.6.0.2   r:/v2/*#033[0m
Nov 27 14:02:12 172.24.0.1 ui[1565]: 2018/11/27 14:02:12 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 200 #033[0m|    29.6358ms|   match|#033[44m GET     #033[0m /harbor/projects   r:/harbor/*#033[0m
Nov 27 14:02:13 172.24.0.1 ui[1565]: 2018/11/27 14:02:13 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 200 #033[0m|    29.1843ms|   match|#033[44m GET     #033[0m /api/systeminfo   r:/api/systeminfo#033[0m
Nov 27 14:02:13 172.24.0.1 ui[1565]: 2018/11/27 14:02:13 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 200 #033[0m|     3.8032ms|   match|#033[44m GET     #033[0m /api/users/current   r:/api/users/:id#033[0m
Nov 27 14:02:13 172.24.0.1 ui[1565]: 2018/11/27 14:02:13 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 200 #033[0m|    10.8309ms|   match|#033[44m GET     #033[0m /api/systeminfo/volumes   r:/api/systeminfo/volumes#033[0m
Nov 27 14:02:13 172.24.0.1 ui[1565]: 2018/11/27 14:02:13 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 200 #033[0m|    25.5993ms|   match|#033[44m GET     #033[0m /api/statistics   r:/api/statistics#033[0m
Nov 27 14:02:13 172.24.0.1 ui[1565]: 2018/11/27 14:02:13 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 200 #033[0m|    12.6823ms|   match|#033[44m GET     #033[0m /api/projects   r:/api/projects/#033[0m
Nov 27 14:02:13 172.24.0.1 ui[1565]: 2018/11/27 14:02:13 #033[1;44m[D] [server.go:2619] |     172.24.0.1|#033[42m 200 #033[0m|      500.3µs|   match|#033[44m GET     #033[0m /favicon.ico#033[0m
Nov 27 14:02:14 172.24.0.1 ui[1565]: 2018/11/27 14:02:14 #033[1;44m[D] [server.go:2619] |      127.0.0.1|#033[42m 200 #033[0m|      222.8µs|   match|#033[44m GET     #033[0m /api/ping   r:/api/ping#033[0m
```
but in harbor, i can find foundation image.

**Pull request closed without merging**